### PR TITLE
[Major] Button focus update & related components

### LINKED ISSUE #990: Focus review
Some component's focus can cause problems distinguishing the outline from the Halstack palette of colors. Others are implemented with box-shadow instead of outline, which can't have an offset to distinguish the focus styling, or they basically don't display the focus when they have been clicked (Switch and Checkbox) and others yes (Button, Text-Input, ...), etc. To establish a uniform focus styling and behaviour we must:

- [x] Update every focus style to (except Date input, Number input, Password input, Text input, Textarea, Select and Accordion):
```
outline: 2px solid #0095ff;
outline-offset: 2px;
```
- [x] Update Switch and Checkbox focus behaviour. They must display the focus when they have been clicked (this also includes the label `onClick`).
- [x] #987.
- [x] Update every storie that has been affected by all these changes.
- [ ] Document all these changes and bring them into the design libraries.

All these changes have force to update other components spacing (here are more details about it #991).

### PR DESCRIPTION
**Pull request by GomezIvann**
Fix #990 and #991.

button styling

And also:

- File Input styles updated to fit the new styling.
- Toggle Group styles are updated to fit the new styling. Also, the component changes its `background-color` from `#fafafa` to `transparent`.

### CODE REVIEW

**Comment by GomezIvann** [comment]
Thanks for everything @aweell, enjoy your next chapter! 🙂 

awell

File: lib/src/file-input/FileInput.tsx [hunk 1]
```diff
@@ -308,6 +310,7 @@ const Label = styled.label`
   font-size: ${(props) => props.theme.labelFontSize};
   font-weight: ${(props) => props.theme.labelFontWeight};
   line-height: ${(props) => props.theme.labelLineHeight};
+  ${(props) => !props.helperText && `margin-bottom: 0.5rem;`}
 `;
```
Comment (raquelarrojo) [inline lib/src/file-input/FileInput.tsx:314]: Same here. In the specs these margins are shown as 0.25 rem. Is this change really needed and the specs should be updated?
image

File: lib/src/file-input/FileInput.tsx [hunk 2]
```diff
@@ -316,23 +319,31 @@ const HelperText = styled.span`
...
   flex-direction: ${(props) => (props.mode === "filedrop" ? "row" : "column")};
-  ${(props) => props.mode === "dropzone" && "justify-content: center; padding: 1rem;"};
+  ${(props) => (props.mode === "filedrop" ? "column-gap: 0.875rem;" : "row-gap: 0.875rem;")}
+  ${(props) => props.mode === "dropzone" && "justify-content: center; padding: 1rem; min-height: 160px;"};
   align-items: center;
-  height: ${(props) => (props.mode === "filedrop" ? "48px" : "160px")};
   width: 320px;
-
+  padding: 0.375rem;
   box-shadow: 0 0 0 2px transparent;
```
Comment (raquelarrojo) [inline lib/src/file-input/FileInput.tsx:341]: Same here. Padding and also border radius are different in the guidelines.

File: lib/src/toggle-group/ToggleGroup.tsx [hunk 3]
```diff
@@ -116,6 +117,7 @@ const Label = styled.label`
   font-style: ${(props) => props.theme.labelFontStyle};
   font-weight: ${(props) => props.theme.labelFontWeight};
   line-height: ${(props) => props.theme.labelLineHeight};
+  margin-bottom: ${(props) => !props.helperText && props.theme.containerMarginTop};
 `;
```
Comment (raquelarrojo) [inline lib/src/toggle-group/ToggleGroup.tsx:121]: Same here. In the specs these margins are shown as 0.25 rem. Is this change really needed and the specs should be updated?
image

File: lib/src/toggle-group/ToggleGroup.tsx [hunk 4]
```diff
@@ -144,6 +147,7 @@ const ToggleGroup = styled.div`
 const OptionsContainer = styled.div`
   display: flex;
   flex-direction: row;
+  column-gap: 0.375rem;
   width: max-content;
```
Comment (raquelarrojo) [inline lib/src/toggle-group/ToggleGroup.tsx:151]: Space between toggles is 4px in guidelines.

File: lib/src/toggle-group/ToggleGroup.tsx [hunk 5]
```diff
@@ -152,15 +156,13 @@ const OptionsContainer = styled.div`
   border-radius: ${(props) => props.theme.containerBorderRadius};
   border-color: ${(props) => props.theme.containerBorderColor};
   background-color: ${(props) => props.theme.containerBackgroundColor};
-  padding: 4px;
-  margin-top: ${(props) => props.theme.containerMarginTop};
+  padding: 0.375rem;
 `;
```
Comment (raquelarrojo) [inline lib/src/toggle-group/ToggleGroup.tsx:160]: Space between toggles is 4px in guidelines.
